astropy/coordinates/sky_coordinate.py

```diff
@@ -206,12 +208,36 @@ class or the corresponding string alias.  The frame classes that are built in
     # info property.
     info = SkyCoordInfo()
 
+    # A counter to keep track of the (global) number of times SkyCoord has been
+    # initialized
+    _init_counter = 0
+
     def __init__(self, *args, copy=True, **kwargs):
 
         # these are frame attributes set on this SkyCoord but *not* a part of
         # the frame object this SkyCoord contains
         self._extra_frameattr_names = set()
 
+        # Increment the SkyCoord init counter:
+        SkyCoord._init_counter += 1
+
+        # If there is a threshold in the config, check the number of times
+        # SkyCoord has been initialized and possibly issue a warning
+        from . import conf
+        _counter_lim = conf.skycoord_init_counter_warn_threshold
+        if _counter_lim > 0 and SkyCoord._init_counter > _counter_lim:
+            url = "http://docs.astropy.org/en/latest/coordinates/skycoord.html#array-operations"
+            warnings.warn("We noticed that you have initialized SkyCoord more "
```

Comment:
This is not filtering on scalar inits, so the user may well have made 100 (or 10000) array SkyCoords, so this message will be confusing.